packages/nimble-components/src/table/specs/README.md

```diff
@@ -151,14 +149,16 @@ _Functions_
 
 _Events_
 
-Placeholder
+-   `action-menu-opening` - An event that is emitted immediately prior to the action menu opening. This can be used to update the items in the menu so that they are in the correct state for the row(s) the menu is associated with. The event details include the following:
```

Comment:
Html spec for popovers just got approved! The event for "I'm about to open but haven't opened yet" is "popoveropen": https://open-ui.org/components/popup.research.explainer#events

So for our custom event I'd keep the hyphens but suggest we go `action-menu-open`

Reply:
I'm guessing this naming convention should also be applied to the `nimble-menu-button`? Should the new event I'm proposing to add on that component be `show` rather than `opening`? It already has an event named `open-change` that fires right after the menu is opened or closed. Do you think there is a better name for the existing event? I don't see an event in the popup spec that aligns with our existing event. It would be a breaking change, but I don't think there are any usages of it (at least in SLE).

Reply:
turns out the popover merge jumped the gun, was reverted, and [reopened](https://github.com/whatwg/html/pull/8717) 😅

Also the explainer I linked seems to be out of date with the current changes, it looks like the current idea (most likely to merge I think) is a single [toggle event](https://github.com/openui/open-ui/issues/607#issuecomment-1319114654) instead of separate show/hide.

That seems more in line with the "open-change" on menu-button. 

(haven't fully responded to your qs yet, just posting some updates)

Reply:
Looks like it's settling on `beforetoggle` and toggle with `newState` and `oldState` in details: https://github.com/whatwg/html/pull/8717#issuecomment-1386178070

Wanna copy `beforetoggle` for the menu-button? Seems like reasonable alignment (since eventually we will probably use that feature).

Guess we still want to scope the name in table since there will be other "toggley" things happening, so `action-menu-beforetoggle`?

Reply:
`beforetoggle` on the menu-button and `action-menu-beforetoggle` seem like reasonable names to me. I assume that means we'd also want to fire the events right before they close too. Is that your thought as well? I'm guessing most clients will only care about the menu opening, but it seems odd to have a `beforetoggle` event that only fires when toggling in one direction (i.e. from closed to open).

Reply:
>  assume that means we'd also want to fire the events right before they close too
yep! makes sense to do it in both toggle directions 👍

Reply:
I've updated all the table-related specs. I plan to update the README for the menu-button when I make the changes to that component.